Makefile

```diff
@@ -0,0 +1,79 @@
+########################################################
+
+NAME := ftl_installer
+
+# VERSION file provides one place to update the software version.
+VERSION := $(shell cat VERSION)
+
+ifndef FTL_PYTHON
+FTL_PYTHON=python
+endif
+
+PYVER=$(shell $(FTL_PYTHON) -c "import sys;t='{v[0]}.{v[1]}'.format(v=list(sys.version_info[:2]));sys.stdout.write(t)")
+
+__init__.py: ftl_installer/__init__.py.in
+	sed "s/%VERSION%/$(VERSION)/" $< > ftl_installer/__init__.py
+
+setup.py: setup.py.in
+	sed "s/%VERSION%/$(VERSION)/" $< > setup.py
+
+clean:
+	@find . -type f -regex ".*\.py[co]$$" -delete
+	@find . -type f \( -name "*~" -or -name "#*" \) -delete
+	@rm -fR venv build dist $(NAME).egg-info
+	@rm -fR setup.py ftl_installer/__init__.py
+
+venv: clean __init__.py setup.py
+	@echo "#############################################"
+	@echo "# Creating a python-$(PYVER) virtualenv"
+	@echo "#############################################"
+	$(FTL_PYTHON) -m virtualenv venv
+	. venv/bin/activate && pip install -r requirements.txt
+	. venv/bin/activate && python setup.py develop
+
+install: __init__.py setup.py
+	python setup.py install
+	pip install -r test-requirements.txt
+	pip install coveralls
+
+nosetests: __init__.py setup.py
+	@echo "#############################################"
+	@echo "# Running Unit Tests"
+	@echo "#############################################"
+	python setup.py nosetests
+
+pep8: __init__.py setup.py
+	@echo "#############################################"
+	@echo "# Running PEP8 Compliance Tests"
+	@echo "#############################################"
+	pep8 --max-line-length=120 --exclude="tests/*,venv/*" --ignore=E501,E121,E124 $(NAME)/
+
+pyflakes: __init__.py setup.py
+	@echo "#############################################"
+	@echo "# Running Pyflakes Sanity Tests"
+	@echo "# Note: most import errors may be ignored"
+	@echo "#############################################"
+	pyflakes $(NAME)/
+
+ci-nosetests: __init__.py setup.py
```

Comment:
With `make ci` being the main entry point for these I don't think we'd want to run the venv for each ci- target